### PR TITLE
My Second Code Review

### PR DESCRIPTION
**Pull request by TonyTheMacaroni1**
This program lets you create weekly or reoccurring events and records days that you are not available and plan events based on your free time.
The Program was available on git on time
The program compiles with no errors
The program is easy to read and is well organized with enough comments to understand what each part specifically means.

Some of the suggestions that I made are:
Adding whitespaces in long data classes for readability
renaming str variable to formattedOutput for easier understandability of the getFormattedDate() function
changing if - else to when on some functions to simplify code https://kotlinlang.org/docs/control-flow.html

What I have learned from this code review is:
How to add custom events to an existing calendar
How to change if - else expressions to when expressions for much simpler codes
How to implement swipe gestures into an app

### CODE REVIEW

File: app/src/main/java/uc/edu/klopfsea/groupscheduler/dto/NewEventDto.kt [hunk 1]
```diff
@@ -1,3 +1,6 @@
 package uc.edu.klopfsea.groupscheduler.dto
 
-data class NewEventDto(var newEventName: String = "", var hour: Int = 0, var minute: Int = 0, var newEventId: String = "")
\ No newline at end of file
+data class NewEventDto(var newEventName: String = "",
```
Comment (TonyTheMacaroni1) [inline app/src/main/java/uc/edu/klopfsea/groupscheduler/dto/NewEventDto.kt:3]: Added whitespaces for easier readability of functions

File: app/src/main/java/uc/edu/klopfsea/groupscheduler/dto/PlannedEvent.kt [hunk 2]
```diff
@@ -4,4 +4,11 @@ package uc.edu.klopfsea.groupscheduler.dto
...
-data class PlannedEvent(var eventName: String? = "", var address: String = "", var plannedTime: String = "", var notes: String = "", var eventDate: String = "", var addresses: ArrayList<Address> = ArrayList<Address>(), var city: String = "", var state: String? = "", var plannedEventId: String = "")
\ No newline at end of file
+data class PlannedEvent(var eventName: String? = "",
+                        var address: String = "",
+                        var plannedTime: String = "",
+                        var notes: String = "",
+                        var eventDate: String = "",
+                        var addresses: ArrayList<Address> = ArrayList<Address>(),
+                        var city: String = "", var state: String? = "",
+                        var plannedEventId: String = "")
```
Comment (TonyTheMacaroni1) [inline app/src/main/java/uc/edu/klopfsea/groupscheduler/dto/PlannedEvent.kt:14]: Added whitespaces for easier readability of functions

File: app/src/main/java/uc/edu/klopfsea/groupscheduler/dto/Schedule.kt [hunk 3]
```diff
@@ -1,3 +1,7 @@
 package uc.edu.klopfsea.groupscheduler.dto
 
-data class Schedule(var day: String? = "", var date: String? = "", var time: String? = "", var scheduleId: String = "", var planEvent: ArrayList<PlannedEvent> = ArrayList<PlannedEvent>())
\ No newline at end of file
+data class Schedule(var day: String? = "",
+                    var date: String? = "",
+                    var time: String? = "",
+                    var scheduleId: String = "",
+                    var planEvent: ArrayList<PlannedEvent> = ArrayList<PlannedEvent>())
```
Comment (TonyTheMacaroni1) [inline app/src/main/java/uc/edu/klopfsea/groupscheduler/dto/Schedule.kt:7]: Added whitespaces for easier readability of functions

File: app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt [hunk 4]
```diff
@@ -17,31 +17,28 @@ data class Event(
...
-            "After $time $period"
-        else if (t == 1)
-            "Before $time $period"
-        else
-            "All Day"
+        timeSpecification = when (t) {
+            0 -> "After $time $period"
+            1 -> "Before $time $period"
+            else -> "All Day"
+        }
```
Comment (TonyTheMacaroni1) [inline app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt:24]: Changed if-else expressions to when for simpler code

File: app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt [hunk 5]
```diff
@@ -17,31 +17,28 @@ data class Event(
...
     }
 
     // formats output
     fun getFormatedDate(): String {
         var s = date.split('-')
-        var str = ""
+        var formattedOutput = ""
```
Comment (TonyTheMacaroni1) [inline app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt:30]: renamed variable str to formattedOutput for easier understandability

File: app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt [hunk 6]
```diff
@@ -17,31 +17,28 @@ data class Event(
...
-        else
-            "th"
-        return str
+        formattedOutput += when (d) {
+            1 -> "st"
+            2 -> "nd"
+            3 -> "rd"
+            else -> "th"
+        }
+        return formattedOutput
```
Comment (TonyTheMacaroni1) [inline app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt:41]: Changed if-else expressions to when for simpler code

File: app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt:30]: I'd go one step further.  What is being outputted here?  Use that to name the variable.

**Comment by discospiff** [comment]
Descriptive variables are a good idea.

Formatting and  control logic (if test) style: a lot of this is programmer's preference, and one format versus another doesn't necessarily reduce technical debt.  So, best to leave it to the group on what formatting they prefer: what they originally have, or what is proposed in this review.

Also note that the IDE can automatically reformat code for you.  You can do that without going through the overhead to merge a branch.

**Pull request closed without merging**